### PR TITLE
Update the universal build with framework dependent package for dotnet sdk

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

Update the universal build to also build the framework package needed for the dotnet sdk container image.

## PR Context  

We build the package in individual builds. This change brings over the steps in the universal build.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
The changes were verified using a private branch on AzDevOps.

File: tools/releaseBuild/azureDevOps/templates/dotnetsdk-fxdependent.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      downloadType: specific
+      itemPattern: |
+        **/*fxd-dotnetsdk.zip
+        **/*fxd-dotnetsdk.tar.gz
+      downloadPath: '$(System.ArtifactsDirectory)'
+    condition: succeeded()
+
+  - powershell: |
+      Get-ChildItem '$(System.ArtifactsDirectory)\release'
+    displayName: List downloaded artifacts
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/dotnetsdk-fxdependent.yml]: Can we use `Capture` as the first word of diagnostics tasks like this?

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -3109,9 +3110,10 @@ function New-DotnetSdkContainerFxdPackage {
     Write-Log "fxdPackagePath: $FxdPackagePath"
 
     $packageName = "powershell-$Version-$packageNamePlatform-fxd-dotnetsdk$packageNameExtension"
+    $destinationPackageFullName = Join-Path $DestinationPath $packageName
 
     ## Get fxdependent package path
-    $fxdPackage = (Get-ChildItem $FxdPackagePath -Recurse -Filter $basePackagePattern).FullName
+    $fxdPackage = (Get-ChildItem $FxdPackagePath -Recurse -Filter $basePackagePattern).FullName | Select-Object -First 1
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: NIT:  I always find this odd.  Seems more logical to use `-ExpandProperty` to me

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -3090,7 +3090,8 @@ function New-DotnetSdkContainerFxdPackage {
     [CmdletBinding(SupportsShouldProcess = $true)]
     param(
         [Parameter(Mandatory)] $FxdPackagePath,
-        [Parameter(Mandatory)] $ReleaseTag
+        [Parameter(Mandatory)] $ReleaseTag,
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: It's good to validate this.  You can find the regex in other places in the code.  It helps if someone messes up passing this parameter into the script.

Consider similar issues on the other parameters.

**Review by TravisEz13 — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/templates/dotnetsdk-fxdependent.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/dotnetsdk-fxdependent.yml]: Fixed

File: tools/packaging/packaging.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Fixed.

File: tools/packaging/packaging.psm1 (hunk 3 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Fixed

**Review by TravisEz13 — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**